src/sql-parser/src/parser.rs

```diff
@@ -5883,7 +6127,7 @@ impl<'a> Parser<'a> {
 
     /// Parse a `DECLARE` statement, assuming that the `DECLARE` token
     /// has already been consumed.
-    fn parse_declare(&mut self) -> Result<Statement<Raw>, ParserError> {
+    fn parse_declare(&mut self) -> Result<Statement<Raw>, ParserStatementError> {
```

Comment:
Ditto.

Reply:
Same reason: the internal nested `parse_statement` returns a `ParserStatementError`.